services/ingestion/etl/2014tc16rftn002/xls/src/events/onParseXLS.js

```diff
@@ -0,0 +1,143 @@
+import AWS from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+import XLSX from 'xlsx';
+
+// ETL utilities.
+import ensureExtensions from '@eubfr/lib/etl/ensureExtensions';
+import extractMessage from '@eubfr/lib/etl/extractMessage';
+import handleError from '@eubfr/lib/etl/handleError';
+
+import MessengerFactory from '@eubfr/logger-messenger/src/lib/MessengerFactory';
+import { STATUS } from '@eubfr/logger-messenger/src/lib/status';
+
+import transformRecord from '../lib/transform';
+
+export const handler = async (event, context) => {
+  const { BUCKET, REGION, STAGE } = process.env;
+
+  if (!BUCKET || !REGION || !STAGE) {
+    throw new Error(
+      'BUCKET, REGION and STAGE environment variables are required!'
+    );
+  }
+
+  try {
+    const snsMessage = extractMessage(event);
+    const { key } = snsMessage.object;
+
+    if (!ensureExtensions({ file: key, extensions: ['.xls', '.xlsx'] })) {
+      throw new Error('XLS or XLSX file expected for this ETL.');
+    }
+
+    const messenger = MessengerFactory.Create({ context });
+    const s3 = new AWS.S3();
+
+    await messenger.send({
+      message: {
+        computed_key: key,
+        status_message: 'Start parsing XLS...',
+        status_code: STATUS.PARSING,
+      },
+      to: ['logs'],
+    });
+
+    // Get file
+    const readStream = s3
+      .getObject({ Bucket: snsMessage.bucket.name, Key: key })
+      .createReadStream();
+
+    return new Promise((resolve, reject) => {
+      // Put data in buffer
+      const buffers = [];
+      readStream.on('data', data => {
+        buffers.push(data);
+      });
+
+      readStream.on('error', async e =>
+        handleError(
+          { messenger, key, statusCode: STATUS.ERROR },
+          { error: e, callback: reject }
+        )
+      );
+
+      // Manage data
+      readStream.on('end', async () => {
+        let dataString = '';
+
+        // Parse file
+        const buffer = Buffer.concat(buffers);
+        const workbook = XLSX.read(buffer);
+        const sheetNameList = workbook.SheetNames;
+        const parsedRows = XLSX.utils.sheet_to_json(
+          workbook.Sheets[sheetNameList[0]]
+        );
+
+        // Remove first row "LIST OF OPERATIONS APPROVED UNDER THE FIRST CALL FOR PROJECTS (2016/2017)".
```

Comment:
😆

Reply:
Yep, many sheets contain unnecessary data, and I'm thinking of our future-selves reasoning behind removing data :D